framework/wazuh/rbac/orm.py

```diff
@@ -1477,52 +1479,52 @@ def add_policy(self, name: str, policy: dict, policy_id: int = None, created_at:
-                                policy_id = max_id_reserved + 1
+            # Check if the policy body is valid
+            if not policy or not json_validator(policy) or len(policy.keys()) != 3:
+                return SecurityError.INVALID
 
-                        except (TypeError, AttributeError):
-                            pass
-                        self.session.add(Policies(name=name, policy=json.dumps(policy), policy_id=policy_id,
-                                                  created_at=created_at, resource_type=resource_type))
-                        self.session.commit()
-                        return True
-                    else:
-                        return SecurityError.INVALID
+            # To add a policy it must have the keys actions, resources, effect
+            if 'actions' in policy.keys() and 'resources' in policy.keys() and 'effect' in policy.keys():
+                # The keys actions and resources must be lists and the key effect must be str
+                if isinstance(policy['actions'], list) and isinstance(policy['resources'], list) \
+                        and isinstance(policy['effect'], str):
+                    # Regular expression that prevents the creation of invalid policies
+                    regex = r'^[a-zA-Z_\-*]+:[a-zA-Z0-9_\-*]+([:|&]{0,1}[a-zA-Z0-9_\-\/.*]+)*$'
+                    for action in policy['actions']:
+                        if not re.match(regex, action):
+                            return SecurityError.INVALID
+                    for resource in policy['resources']:
+                        if not re.match(regex, resource):
+                            return SecurityError.INVALID
+
+                    try:
+                        if not check_default:
+                            policies = sorted([p.id for p in self.get_policies()]) or [0]
+                            policy_id = max(filter(lambda x: not(x > cloud_reserved_range), policies)) + 1
+
+                        elif check_default and \
+                                self.session.query(Policies).order_by(desc(Policies.id)
+                                                                      ).limit(1).scalar().id < max_id_reserved:
+                            policy_id = max_id_reserved + 1
+
+                    except (TypeError, AttributeError):
+                        pass
+
+                    self.session.add(Policies(name=name, policy=json.dumps(policy), policy_id=policy_id,
+                                              created_at=created_at, resource_type=resource_type))
+                    self.session.commit()
+                    return True
                 else:
                     return SecurityError.INVALID
             else:
                 return SecurityError.INVALID
         except IntegrityError:
             self.session.rollback()
-            return SecurityError.ALREADY_EXIST
+            return SecurityError.CONSTRAINT_ERROR
```

Comment:
This error includes too long names as well. We should handle these separately.

Reply:
This will reviewed and improved in another forthcoming issue